pravega-client-examples/src/main/java/io/pravega/example/secure/SecureBatchReader.java

```diff
@@ -0,0 +1,228 @@
+
+        final String uriString = cmd.getOptionValue("uri") == null ? Constants.DEFAULT_CONTROLLER_URI : cmd.getOptionValue("uri");
+        final URI controllerURI = URI.create(uriString);
+
+        final String truststorePath =
+                cmd.getOptionValue("truststore") == null ? Constants.DEFAULT_TRUSTSTORE_PATH : cmd.getOptionValue("truststore");
+        final boolean validateHostname = cmd.getOptionValue("validatehost") == null ? false : true;
+
+        final String username = cmd.getOptionValue("username") == null ? Constants.DEFAULT_USERNAME : cmd.getOptionValue("username");
+        final String password = cmd.getOptionValue("password") == null ? Constants.DEFAULT_PASSWORD : cmd.getOptionValue("password");
+
+        SecureBatchReader reader = new SecureBatchReader(scope, stream, controllerURI,
+                truststorePath, validateHostname,
+                username, password);
+        reader.read();
+    }
+
+    private static Options getOptions() {
+        final Options options = new Options();
+        options.addOption("s", "scope", true, "The scope name of the stream to read from.");
+        options.addOption("n", "stream", true, "The name of the stream to read from.");
+        options.addOption("u", "uri", true, "The URI to the controller in the form tls://host:port");
+        options.addOption("t", "truststore", true,
+                "The location of .pem truststore file in the file system to use by this application process.");
+        options.addOption("v", "validatehost", false, "Whether to verify server's hostname.");
+        options.addOption("a", "username", true,
+                "The account username to use by the client for authenticating to the server.");
+        options.addOption("p", "password", true,
+                "The account password to use by the client for authenticating to the server.");
+        return options;
+    }
+
+    private static CommandLine parseCommandLineArgs(Options options, String[] args) throws ParseException {
+        CommandLineParser parser = new DefaultParser();
+        CommandLine cmd = parser.parse(options, args);
+        return cmd;
+    }
+
+    public void run() {
+        try (BatchClientFactory batchClient = BatchClientFactory.withScope(scope, clientConfig)) {
+            System.out.println("Done creating batchClient for " + scope + "/" + stream);
+            ArrayList<SegmentRange> ranges = Lists.newArrayList(batchClient.getSegments(Stream.of(scope, stream), StreamCut.UNBOUNDED, StreamCut.UNBOUNDED).getIterator());
+            readFromRanges(batchClient, ranges);
+            System.out.println("Done reading ranges of size: " + String.valueOf(ranges.size()));
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    private int readFromRanges(BatchClientFactory batchClient, List<SegmentRange> ranges) {
```

Comment:
Since this code is for demo and samples, it should be as simple as it can be. Try to fold all of the major business logic in one method like is done here: 

https://github.com/pravega/pravega-samples/blob/ed9d0f883e07fcbc917c1ac7ce702ef3e31d84df/pravega-client-examples/src/main/java/io/pravega/example/secure/SecureReader.java#L72

It might help to think of someone who is totally new to Pravega and how he'd make sense of this code.

Reply:
Yes Ravi